pinot-plugins/pinot-file-system/pinot-s3/src/test/java/org/apache/pinot/plugin/filesystem/S3PinotFSTest.java

```diff
@@ -165,17 +165,16 @@ public void testListFilesInFolderNonRecursive()
     for (String fileName : originalFiles) {
       createEmptyFile(folder, fileName);
     }
-    // Files in sub folders should be skipped.
+
     createEmptyFile(folder + DELIMITER + "subfolder1", "a-sub-file.txt");
     createEmptyFile(folder + DELIMITER + "subfolder2", "a-sub-file.txt");
 
-    String[] actualFiles = _s3PinotFS.listFiles(URI.create(String.format(FILE_FORMAT, SCHEME, BUCKET, folder)), false);
-    Assert.assertEquals(actualFiles.length, originalFiles.length);
+    String[] expectedFiles = new String[]{"a-list-2.txt", "b-list-2.txt", "c-list-2.txt", "subfolder1", "subfolder2"};
 
-    actualFiles = Arrays.stream(actualFiles).filter(x -> x.contains("list-2")).toArray(String[]::new);
-    Assert.assertEquals(actualFiles.length, originalFiles.length);
+    String[] actualFiles = _s3PinotFS.listFiles(URI.create(String.format(FILE_FORMAT, SCHEME, BUCKET, folder)), false);
+    Assert.assertEquals(actualFiles.length, expectedFiles.length);
 
-    Assert.assertTrue(Arrays.equals(Arrays.stream(originalFiles)
+    Assert.assertTrue(Arrays.equals(Arrays.stream(expectedFiles)
```

Comment:
Let's update this to use `Assert.assertEquals` instead (which handles array equality checks appropriately) so that we get better error messages in failure scenarios.

Reply:
Sounds good